### DISCUSSION TITLE
styling parent div of ui.row element

### DISCUSSION THREAD

**Discussion by monkeyfab** [zauberzeug/nicegui#705] [Q&A] [answered]
### Question

Hello there, how to remove gaps between ui.row() elements?

**Comment by falkoschindler** [comment]
You have multiple options.

- CSS:
    ```py
    with ui.row().style('gap: 0'):
        ui.label('A')
        ui.label('B')
    ```

- Tailwind classes:
    ```py
    with ui.row().classes('gap-0'):
        ui.label('A')
        ui.label('B')
    ```

- the new `tailwind` API (coming in NiceGUI 1.2.4):
    ```py
    with ui.row() as row:
        row.tailwind.gap('0')
        ui.label('A')
        ui.label('B')
    ```
    or:
    ```py
    with ui.row() as row:
        row.tailwind('gap-0')
        ui.label('A')
        ui.label('B')
    ```

**Comment by monkeyfab** [comment]
Gee, you're fast! 

I probably described it badly, I meant gaps between ui.row() elements. 

The rows sit in a div with class 'nicegui-content' and I'd like to add class 'gap-0' to that div. 

I have something like this: 

```py
for machine_id, machine_ip in machine_ips.items(): 
    self.row_element = ui.row().classes('items-center')
    with self.row_element:
    for name, ui_elem_type in self.column_defs.items(): 
        ...
```

I try to shrink the gaps to make it more compact.    

That said...huge thanks for this great piece of software, I've been using Tkinter to write simple desktop apps for us, looking for a way to write a web GUI python. 

Now I'm playing with NiceGUI for a few days and it's just so good. The ability to use normal python code with all the modules is a total game changer. 

The simplicity and clarity of it...the ui.timer to update the model... 

You guys are amazing! 

Best regards 

Piotr

**Reply by monkeyfab** [reply]
I packed the rows into a ui.elelment('div').classes('gap-0'), and there is no gap between them, sort of a hack, but works.

**Reply by falkoschindler** [reply]
Ok, I understand. Wrapping your rows into another parent container is a totally valid approach. 👍 
I would probably use a `ui.column().classes('gap-0')` to get the flex layout behavior. But `ui.element('div')` is also ok. This way you probably don't need to set `gap-0` because a div has not gap by default.

Anyway, glad to hear that you enjoy using NiceGUI. 😊

**Comment by falkoschindler** [comment, marked as answer]
To change the gap of the page content, you can manipulate the `client.content`. To get the client, you can either use a page function or the `globals` module:

```py
from nicegui import Client, ui

@ui.page('/')
def main_page(client: Client):
    client.content.classes('gap-0')
    ui.label('A')
    ui.label('B')

ui.run()
```

```py
from nicegui import globals, ui

globals.get_client().content.classes('gap-0')
ui.label('A')
ui.label('B')

ui.run()
```

**Reply by monkeyfab** [reply]
Yeah! Like butter! Thanks!
